components/Post/PostContent.tsx

```diff
@@ -2,20 +2,28 @@ import config from "config";
 import toHumanReadableDate from "helpers/toHumanReadableDate";
 import { PostDocument } from "interfaces";
 import Head from "next/head";
+import { useEffect, useRef } from "react";
+
+import renderCopyButtons from "../../helpers/renderCopyButtons";
 
 interface Props {
   post: PostDocument;
 }
 
-const getFirstParagraph = (str: string) => {
-  const openedElement = str.indexOf("<p>") + 3;
-  const closedElement = str.indexOf("</p>");
+const getFirstParagraph = (string: string) => {
+  const openedElement = string.indexOf("<p>") + 3;
+  const closedElement = string.indexOf("</p>");
 
-  return str.slice(openedElement, closedElement);
+  return string.slice(openedElement, closedElement);
 };
 
 const PostContent = ({ post }: Props) => {
   const description = getFirstParagraph(post.content);
+  const document = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    return renderCopyButtons(document);
```

Comment:
@ntorbinskiy I'm not sure you need a return statement in here, do you?

Reply:
@1alexvash 
No, I need it, because I return a teardown method that'll remove all EventListeners

Reply:
@ntorbinskiy usually the cleanup looks like this:

```ts
  useEffect(() => {
    attachEventListeners();
    return () => removeEvenetListeners()
  }, [post.content])
```

🤔